server/schema/user/user.datasources.js

```diff
@@ -178,32 +178,33 @@ const updateDetails = (id, fields, session, authToken, mid) =>
     { new: true }
   );
 
-const getCustomClaims = async (id) => {
+const getCustomClaims = async (email) => {
   try {
-    const _user = await UserModel.findById(id, 'email');
-    if (!_user) {
+    const _fbUser = await admin.auth().getUserByEmail(email);
+    if (!_fbUser) {
       throw APIError('NOT_FOUND', null, { reason: 'The requested user does not exist.' });
     }
-    const _fbUser = await admin.auth().getUserByEmail(_user.email);
     const userRecord = await admin.auth().getUser(_fbUser.uid);
     return userRecord.customClaims;
   } catch (error) {
     throw FirebaseAuthError(error, { reason: "Cannot find user's roles" });
   }
 };
-const updateRoles = async (id, roles) => {
+const updateRoles = async (email, roles) => {
   try {
-    const _user = await UserModel.findById(id);
-    if (!_user) {
+    console.log('Update roles called');
+    const _fbUser = await admin.auth().getUserByEmail(email);
+    console.log(_fbUser);
+    if (!_fbUser) {
       throw APIError('NOT_FOUND', null, { reason: 'The requested user does not exist.' });
     }
-
-    const _fbUser = await admin.auth().getUserByEmail(_user.email);
     await admin.auth().setCustomUserClaims(_fbUser.uid, {
       ..._fbUser.customClaims,
       roles,
     });
-    return _user;
+    const customClaims = await getCustomClaims(email);
+    console.log(customClaims);
+    return customClaims.roles;
```

Comment:
2 Questions here
- As a consumer of the API/function, this function returns the list of roles of the user but does not return the userID. Shouldn't the function also return the user identifier, without that, the roles are not much useful
- Why do we need to query DB again in L205? Given that the save operation was a success in L201, we can be assured it is saved to DB and available to the server when the user makes another request to fetch articles or other information. This is an almost redundant DB operation

Reply:
@DesignrKnight 
   -> Yup as a consumer of a API having a Id with the roles would be more informative, hence now I added email id in 
        the response of this function 
   -> Line no 205 was completely redundant hence it was removed.